analysis/src/python/evaluation/tools/hyperstyle/model/report.py

```diff
@@ -38,17 +57,25 @@ def to_str(self) -> str:
 
 
 @dataclass_json
-@dataclass(frozen=True)
-class HyperstyleReport(QualityReport):
+@dataclass(frozen=True, eq=True)
+class HyperstyleReport(QualityReport, BaseReport):
     issues: List[HyperstyleIssue]
 
+    def get_issues(self) -> List[HyperstyleIssue]:
+        return self.issues
+
+    def filter_issues(self, predicate: Callable[[BaseIssue], bool]) -> 'HyperstyleReport':
+        # TODO: recalculate quality after filtering
```

Comment:
How are you going to recalculate it?

Reply:
It is a tip for the future if we need it. When we filter template issues, for example, the quality can change, but for now I use one from the original report =( It is ok for now but if we want to analyze quality, we need to recalculate it